fix: credential hash in the credential_pop (#6)
peppelinux/draft-demarco-oauth-status-attestations: draft-demarco-oauth-status-attestations.md

~~~

The Wallet Instance sends the Status Attestation request to the Credential Issuer.

> The request MUST contain the hash value of the Digital Credential, for which the Status Attestation

Reviewer: ```suggestion
The request MUST contain the base64urlencoded hash value of the Digital Credential, for which the Status Attestation
```

    "exp": 1698834139,
    "jti": "6f204f7e-e453-4dfd-814e-9d155319408c",

>     "credential_format": "vc+sd-jwt",

Author: @peppelinux probably we could remove the credential_format. 
WDYT?

Reviewer: yes, I'm on it

>     "credential_format": "vc+sd-jwt",

Reviewer: ```suggestion
```

Reviewer 2: seems like the `typ` or `cty` claim might be used here as well.

Author: I don't see how having these claims would be useful. How would the Credential Issuer use this information?

Reviewer: yes, let's reduce the information about the payload of the credential for an augmented privacy

Reviewer: @OR13 I have asked to remove the credential format, since we give only the credential hash without disclosing any information about the credential

| **exp** | UNIX Timestamp with the expiration time of the JWT. | {{RFC9126}}, {{RFC7519}} |
| **iat** | UNIX Timestamp with the time of JWT issuance. | {{RFC9126}}, {{RFC7519}} |
| **jti** | Unique identifier for the JWT.  | {{RFC7519}} Section 4.1.7 |
| **credential_hash** | It MUST contain the hash value of the Credential. | this specification |

> | **credential_hash_alg** | The Algorithm used of hashing the Digital Credential. The value SHOULD be set to `S256`. | this specification |

Reviewer: ```suggestion
| **credential_hash_alg** |  The hash value of a Digital Credential is derived by computing the base64url encoded hash of the Digital Credential using one of the hash algorithms listed in the [IANA - Named Information Hash Algorithm Registry](https://www.iana.org/assignments/named-information/named-information.xhtml#hash-alg). Among the hash algorithms, sha-256 is mandatory and MUST be implemented by all systems. | this specification |
```

Reviewer: @OR13 WDYT?^

> | **credential_hash_alg** |  The hash value of a Digital Credential is derived by computing the base64url encoded hash of the Digital Credential. | this specification |

Reviewer: ```suggestion
| **credential_hash_alg** |  The hash value of a Digital Credential, derived by computing the base64url encoded hash of the Digital Credential. | this specification |
```

| **iat** | UNIX Timestamp with the time of JWT issuance. | {{RFC9126}}, {{RFC7519}} |
| **jti** | Unique identifier for the JWT.  | {{RFC7519}} Section 4.1.7 |

> | **credential_format** | The data format of the Credential. Eg: `vc+sd-jwt` for SD-JWT, `vc+mdoc` for ISO/IEC 18013-5 MDOC CBOR [@ISO.18013-5] | this specification |

Reviewer: ```suggestion
```

| **credential_hash** | It MUST contain the hash value of the Credential. | this specification |

> | **credential_hash_alg** | The Algorithm used of hashing the Digital Credential. The value SHOULD be set to `S256`. | this specification |

Reviewer 2: lets use the IANA named hash registry, and make sha-256 mandatory to implement.

https://www.iana.org/assignments/named-information/named-information.xhtml#hash-alg

Alternatively, we can use the cose registry which includes hash algorithms:

https://www.iana.org/assignments/cose/cose.xhtml#algorithms

the "COSE" style for this kind of thing would be:

```
credential_hash: [alg, hash]
```

Author: Regarding the claim name I prefer the first option. 
With regard to the sha-256, I suggest to make the claim OPTIONAL and if omitted we could say that the default value is sha-256. WDYT?

Reviewer: I would make it REQUIRED since I love thing explicit. If a day SHA-256 would be vulnerable and another specification woudl extend this one bringin an updated set of algs, the implementation would then always specify the alg making this aspect not ambiguous

Reviewer: @OR13 I moved your comment in a [separate issue](https://github.com/peppelinux/draft-demarco-oauth-status-attestations/issues/16), to unblock this PR related to the hash 

A PR wil be create asap, when this PR will be merged

Reviewer: See my change suggestion in the form of comment

## Credential Issuer Metadata

The Credential Issuers that uses the Status Attestations MUST include in their
OpenID4VCI [@!OpenID.VCI] metadata the claims:

- `status_attestation_endpoint`. REQUIRED. It MUST be an HTTPs URL indicating the endpoint where the Wallet Instances can request Status Attestations.

> - `credential_hash_alg_supported`. REQUIRED. The supported Algorithm used by the Wallet Instance to hash the Digital Credential for which the Status Attestation is requested. The value SHOULD be set to `S256`.

Reviewer: ```suggestion
- `credential_hash_alg_supported`. REQUIRED. The supported Algorithm used by the Wallet Instance to hash the Digital Credential for which the Status Attestation is requested,  using one of the hash algorithms listed in the [IANA - Named Information Hash Algorithm Registry](https://www.iana.org/assignments/named-information/named-information.xhtml#hash-alg).
```

## Credential Issuer Metadata

The Credential Issuers that uses the Status Attestations MUST include in their
OpenID4VCI [@!OpenID.VCI] metadata the claims:

- `status_attestation_endpoint`. REQUIRED. It MUST be an HTTPs URL indicating the endpoint where the Wallet Instances can request Status Attestations.

> - `credential_hash_alg_supported`. REQUIRED. The supported Algorithm used by the Wallet Instance to hash the Digital Credential for which the Status Attestation is requested. The value SHOULD be set to `S256`.

Reviewer 2: SD-JWT uses the IANA registry: https://datatracker.ietf.org/doc/html/draft-ietf-oauth-selective-disclosure-jwt-07#section-5.1.1

Reviewer: moved in the issue https://github.com/peppelinux/draft-demarco-oauth-status-attestations/issues/16

JSON member `status_attestation` set to a JSON Object containing the following
member:

> - `credential_hash_alg`. REQUIRED. The Algorithm used for hashing the Digital Credential to which the Status Attestation is bound. The value SHOULD be set to `S256`.

Reviewer: ```suggestion
- `credential_hash_alg`. REQUIRED.  The hash value of a Digital Credential is derived by computing the base64url encoded hash of the Digital Credential using one of the hash algorithms listed in the [IANA - Named Information Hash Algorithm Registry](https://www.iana.org/assignments/named-information/named-information.xhtml#hash-alg). Among the hash algorithms, sha-256 is recommended and SHOULD be implemented by all systems.
```

> - `credential_hash_alg`. REQUIRED.  The hash value of a Digital Credential is derived by computing the base64url encoded hash of the Digital Credential using one of the hash algorithms listed in the [IANA - Named Information Hash Algorithm Registry](https://www.iana.org/assignments/named-information/named-information.xhtml#hash-alg). Among the hash algorithms, `sha-256` is recommended and SHOULD be implemented by all systems.

Reviewer: ```suggestion
- `credential_hash_alg`. REQUIRED. The hash value of a Digital Credential is derived by computing the base64url encoded hash of the Digital Credential using one of the hash algorithms listed in the [IANA - Named Information Hash Algorithm Registry](https://www.iana.org/assignments/named-information/named-information.xhtml#hash-alg). Among the hash algorithms, `sha-256` is recommended and SHOULD be implemented by all systems.
```
